src/stactools/core/utils/raster_footprint.py

```diff
@@ -13,12 +13,277 @@
+            points.
+
+    Returns:
+        List[Tuple[float, float]]: The list of densified points
+    """
+    points: Any = np.asarray(point_list)
+
+    dxdy = points[1:, :] - points[:-1, :]
+    segment_lengths = np.sqrt(np.sum(np.square(dxdy), axis=1))
+    total_length = np.sum(segment_lengths)
+
+    cum_segment_lengths = np.cumsum(segment_lengths)
+    cum_segment_lengths = np.insert(cum_segment_lengths, 0, [0])
+    cum_interp_lengths = np.arange(0, total_length, densification_length)
+    cum_interp_lengths = np.append(cum_interp_lengths, [total_length])
+
+    interp_x = np.interp(cum_interp_lengths, cum_segment_lengths, points[:, 0])
+    interp_y = np.interp(cum_interp_lengths, cum_segment_lengths, points[:, 1])
+
+    return [(x, y) for x, y in zip(interp_x, interp_y)]
+
+
+class RasterFootprint:
+    """Methods for updating Item geometry with the convex hull of valid raster data."""
+
+    def __init__(
+        self,
+        *,
+        asset_names: List[str] = [],
+        precision: int = DEFAULT_PRECISION,
+        densification_factor: Optional[int] = None,
+        simplify_tolerance: Optional[float] = None,
+        no_data: Optional[int] = None,
+        bands: List[int] = [1],
+        crs: Optional[CRS] = None,
+    ) -> None:
+        self.asset_names = asset_names
+        self.precision = precision
+        self.densification_factor = densification_factor
+        self.simplify_tolerance = simplify_tolerance
+        self.no_data = no_data
+        self.bands = bands
+        self.crs = crs
+
+    def update_geometry_from_asset_footprint(self, item: Item) -> bool:
+        asset_name_and_extent = next(
+            self.data_footprints_for_data_assets(item),
+            None,
+        )
+        if asset_name_and_extent is not None:
+            _, extent = asset_name_and_extent
+            item.geometry = extent
+            return True
+        else:
+            return False
+
+    def data_footprints_for_data_assets(
+        self, item: Item
+    ) -> Iterator[Tuple[str, Dict[str, Any]]]:
+        for name, asset in item.assets.items():
+            if not self.asset_names or name in self.asset_names:
+                href = asset.get_absolute_href()
+                if href is None:
+                    logger.error(
+                        f"Could not determine extent for asset '{name}', missing href"
+                    )
+                else:
+                    extent = self.data_footprint(href)
+                    if extent:
+                        yield name, extent
+                    else:
+                        logger.error(f"Could not determine extent for asset '{name}'")
+
+    def data_footprint(self, href: str) -> Optional[Dict[str, Any]]:
+        with rasterio.open(href) as src:
+            self.transform = src.transform
+            self.shape = src.shape
+            if self.crs is None:
+                self.crs = src.crs
```

Comment:
I'm a little wary of mutating the underlying `RasterFootprint` object here, unless we need to. What do you think about:

```suggestion
            if self.crs is None:
               crs = src.crs
            else
               crs = self.crs
```

Then, just use `crs` (not `self.crs`) later.
```

Reply:
No longer exists in refactor.